src/components/MessageBar/MessageBar.jsx

```diff
@@ -10,35 +19,138 @@ import {
   SendIcon,
 } from './MessageBarStyled';
 
-const MessageBar = () => {
-  const [value, setValue] = useState('');
+const MessageBar = ({
+  countryData,
+  setCountryData,
+  setSubscriptionCountryRooms,
+}) => {
+  const [message, setMessage] = useState('');
+  const [isShowJoinBtn, setIsShowJoinBtn] = useState(true);
+  const userId = useSelector(getUser)?.id;
   const textAreaRef = useRef(null);
+  const stompClient = useStompClient();
+  const { sendMessage } = useWebSocket();
 
-  const isInputNotEmpty = Boolean(value?.trim().length);
+  const isInputNotEmpty = Boolean(message?.trim().length);
 
-  const handleChange = e => {
-    setValue(e.target.value);
+  const handleChange = ({ target: { value } }) => setMessage(value);
+
+  useEffect(() => {
+    if (countryData?.isSubscribed) {
+      setIsShowJoinBtn(false);
+    } else {
+      setIsShowJoinBtn(true);
+    }
+  }, [countryData.isSubscribed]);
+
+  useSubscription(
+    `/countries/${countryData?.country?.name}/messages`,
+    response => {
+      const receivedMessage = JSON.parse(response.body);
+
+      setCountryData(prevCountryData => {
+        const updatedGroupMessages = [
+          ...(prevCountryData.country.groupMessages || []),
+          receivedMessage,
+        ];
+        return {
+          ...prevCountryData,
+          country: {
+            ...prevCountryData.country,
+            groupMessages: updatedGroupMessages,
+          },
+        };
+      });
+      console.log('response message', receivedMessage);
+    }
+  );
+
+  const handleSubmit = e => {
+    e.preventDefault();
+
+    if (!stompClient) {
+      console.error('Cannot send message: WebSocket not connected');
+      return;
+    }
+
+    const dataToSend = {
+      content: message,
+      countryId: countryData?.country.id,
+      senderId: userId,
+    };
+
+    sendMessage(dataToSend);
+
+    setMessage('');
+  };
+
+  const handleJoinClick = async () => {
+    console.log('Joined to current country');
+    const fetchData = async id => {
```

Comment:
It doesn't make sense declaring fetchData func. Just try to extract the code outside of the fetchData.

Reply:
Done